---
title: "Woolfe Lab: Transdifferentiation of neuron"
output:
  html_document:
    theme: cerulean
    toc: true
    toc_depth: 4
    fig_caption: true
    fig_width: 8
    fig_height: 6
author: "Meeta Mistry"
---


```{r setup, echo=FALSE}

# Setup report details
clientname="Brian Wainger"
clientemail="brian.wainger@childrens.harvard.edu"
lablocation="Woolfe Lab"
analystname="Meeta Mistry"
analystemail="mmistry@hsph.harvard.edu"
```

Array analysis for `r clientname` (`r clientemail`) at `r lablocation`. Contact `r analystname` (`r analystemail`) for additional details. Request from client was:
  
> Basically we are comparing expression profiles of three populations of cells and there is a good amount of variability between replicate samples for one of the populations because it is a fibroblast-derived neuronal cell
line. We would like to go a little further with the data than we have so far

## Bioconductor and R libraries used
```{r, libraries, echo=TRUE}

library(arrayQualityMetrics)
library(RColorBrewer)
library(pd.mogene.1.0.st.v1)
library(mogene10sttranscriptcluster.db)
library(oligo)
library(limma)
library(plyr)
library(ggplot2)
library(grid)
library(gridExtra)
library(xtable)
library(sva)
library(png)
```

## Get variables
- get base directory for analyses
- specify data and results directories
- specify column headers used in metadata file


```{r variables, echo=TRUE}
# Setup directory variables
baseDir <- '.'
dataDir <- file.path(baseDir, "data")
metaDir <- file.path(dataDir, "meta")
resultsDir <- file.path(baseDir, "results")
```

## Load the data and metadata
```{r load data}
prevWorkDir = getwd()

setwd("data/")
pd <- read.AnnotatedDataFrame('covars.desc')
data <- read.celfiles(rownames(pData(pd)))
phenoData(data) <- pd

# Convert to factors
pData(data)$celltype <- factor(pData(data)$celltype)
pData(data)$Batch <- factor(pData(data)$Batch)

# restore previous working directory
setwd(prevWorkDir)
```


## Quality control
ArrayQualityMetrics QC report for the raw data can be found [here](./results/report_raw/index.html)

```{r QC, echo=FALSE, eval=FALSE}
arrayQualityMetrics(expressionset=data, intgroup=c('celltype', 'Batch'),
                    outdir='./results/report_raw', force=TRUE,  do.logtransform=TRUE)
                    
```

### Normalization 
The data was pre-processed and normalized using RMA() (Robust Multiarray Analysis) algorithm. RMA is the standard method for performing background subtraction, normalization and averaging. With the processed data we find that signal intensity is more similarly distributed across the arrays. Although, based on the QC report Array #8 appears to be an outlier and will be removed for downstream analysis. Density estimates show no indication of major outliers as the smoothed histograms for each array have similar shapes and ranges. ArrayQualityMetrics QC report for the normalized data can be found [here](./results/report_rma/index.html)

```{r qcnorm, echo=FALSE, eval=FALSE}
arrayQualityMetrics(expressionset=data.rma, intgroup=c('celltype', 'Batch'),
                    outdir='./results/report_rma', force=TRUE,  do.logtransform=FALSE)
                    
```

```{r rma}
# apply rma
data.rma <- rma(data)

# Remove outliers
preparedData=prepdata(expressionset=data.rma, intgroup=c('celltype', 'Batch'), 
                      do.logtransform=F)
# Boxplots
bo<- aqm.boxplot(preparedData, subsample=10000, outlierMethod= "KS") 
bo.out<-bo@outliers@which

```


### PCA 
Plot all pairwise combinations of the first 5 principal components for the samples. The more similar the samples are, the closer they will cluster in these plots. In the figure below cell type is coded using red=DRG; blue=iNoc; green=MEF.  A second run of induced nociceptor samples were run in the second batch. Here, we see that while the data clusters well into three cell populations in the first and second principal components -  but the batches are evident in PC3. 


```{r calculate_PCA, echo=FALSE, fig.align='center'}
myPca <- prcomp(t(exprs(data.rma)))
pc<-myPca$x

# Plot first factor of interest
tmpPCAData <- as.data.frame(myPca$x[,1:5])
colors <- c("#FF4040", "#6495ED", "#458B00") # to match colors from AQM
plot(tmpPCAData, col=colors[pData(data.rma)[,"celltype"]], pch=19)
```

## Removing batch effects
To remove the effects we will use [ComBat](http://www.bu.edu/jlab/wp-assets/ComBat/Abstract.html) an empirical Bayes framework for adjusting the data for batch effects. EB is borrowing information across all other genes and samples in hopes that this borrowed information will lead to better estimates/stable inferences.

```{r combat, fig.align='center'}
# Load the control annotations
data(mogene10stCONTROL, package="ArrayTools") 

# Remove control probes from data
gene <- data.rma[!(row.names(exprs(data.rma)) %in% mogene10stCONTROL[,1]),]
gene <- gene[,which(colnames(gene) %in% names(bo.out) == F)]


# Model matrix
pd <-pData(gene)
pd$FileName <- NULL
mod<-model.matrix(~celltype, data=pd)

# Batch correction
data.combat <- ComBat(exprs(gene), pd$Batch, mod, numCovs=NULL, par.prior=TRUE,prior.plots=TRUE)
```

### PCA
From the PCA plots we can see that the clustering of iNoc samples by batch has been removed by ComBat. We can proceed using this batch adjusted expression data for differential expression analysis.

```{r, echo=FALSE, fig.align='center'}
# PCA
myPca <- prcomp(t(data.combat))
pc<-myPca$x

# Plot first factor of interest
tmpPCAData <- as.data.frame(myPca$x[,1:5])
colors <- c("#FF4040", "#6495ED", "#458B00") # to match colors from AQM
plot(tmpPCAData, col=colors[pData(data.rma)[,"celltype"]], pch=19)
```

## Differential expression analysis
We applied [limma](http://www.bioconductor.org/packages/release/bioc/html/limma.html) to the data and used a linear modeling approach to identify gene expression changes between the different cell populations. Only celltype was included in the model and we looked at contrasts between the different cell types. 

```{r}
# Get batch corrected data
exprs(gene) <- data.combat

# Fit a linear model
mod<-model.matrix(~celltype -1, data=pd)
fit <- lmFit(gene, mod)

# Define a comtrast model matrix
cont.matrix <- makeContrasts(MEFvsDRG="celltypeMEF-celltypeDRG", MEFvsiNoc="celltypeMEF-celltypeiNoc", 
                             iNocvsDRG="celltypeiNoc-celltypeDRG", levels=mod)

# Extract the linear model fit for the contrasts
fit2  <- contrasts.fit(fit, cont.matrix)
fit2  <- eBayes(fit2)

# Set threshold 
p.cutoff <- 0.001
logfc.cutoff <- 2

# Get gene tables
MEFvsDRG <- topTable(fit2, coef=1, number=nrow(exprs(gene)))
MEFvsiNoc <- topTable(fit2, coef=2, number=nrow(exprs(gene)))
iNocvsDRG <- topTable(fit2, coef=3, number=nrow(exprs(gene)))

```

### Volcano plots
Gene expression changes were assessed by three different comparisons:     

1. mouse embryonic fibrolasts vs dorsal root ganglion
2. mouse embryonic fibroblasts vs induced nociceptor 
3. induce nociceptor vs dorsal root ganglion

The third comparison is where would anticipate to see the least amount of change, under the assumption that the transdifferentiation process was successful. Along the same vein, we would probabaly expect to see a good amount of overlap between the genes identified in the first two comparisons. The volcano plots below highlight the findings from each comparison with significant probes (FDR < 0.001; logFC > 2) represented in turquoise.

```{r volcanoplot, echo=FALSE, fig.align='center', warning=FALSE}

# Highlight genes 
MEFvsDRG$threshold.FDR = as.logical(abs(MEFvsDRG$logFC) > logfc.cutoff & MEFvsDRG$adj.P.Val < p.cutoff)
MEFvsiNoc$threshold.FDR = as.logical(abs(MEFvsiNoc$logFC) > logfc.cutoff & MEFvsiNoc$adj.P.Val < p.cutoff) 
iNocvsDRG$threshold.FDR = as.logical(abs(iNocvsDRG$logFC) > logfc.cutoff & iNocvsDRG$adj.P.Val < p.cutoff)

SigGenes <- c(length(which(MEFvsDRG$threshold.FDR =="TRUE")), length(which(MEFvsiNoc$threshold.FDR =="TRUE")),
                   length(which(iNocvsDRG$threshold.FDR =="TRUE")))

##Construct the plot objects
p1 <- ggplot(data=MEFvsDRG, aes(x=logFC, y=-log10(P.Value), colour=threshold.FDR)) +
  geom_point(alpha=0.4, size=1.75) +
  theme(legend.position = "none",
        plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = rel(1.5)),
        axis.text = element_text(size = rel(1.25))) +
  ggtitle('MEF versus DRG') +
  xlim(c(-10, 10)) + ylim(c(0, 15)) +
  xlab("log2 fold change") + ylab("-log10 p-value")

p2 <- ggplot(data=MEFvsiNoc, aes(x=logFC, y=-log10(P.Value), colour=threshold.FDR)) +
  geom_point(alpha=0.4, size=1.75) +
  theme(legend.position = "none",
        plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = rel(1.5)),
        axis.text = element_text(size = rel(1.25))) +
  ggtitle('MEF versus iNoc') +
  xlim(c(-10, 10)) + ylim(c(0, 15)) +
  xlab("log2 fold change") + ylab("-log10 p-value")

p3 <- ggplot(data=iNocvsDRG, aes(x=logFC, y=-log10(P.Value), colour=threshold.FDR)) +
  geom_point(alpha=0.4, size=1.75) +
  theme(legend.position = "none",
        plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = rel(1.5)),
        axis.text = element_text(size = rel(1.25))) +
  ggtitle('iNoc versus DRG') +
  xlim(c(-10, 10)) + ylim(c(0, 15)) +
  xlab("log2 fold change") + ylab("-log10 p-value")

grid.arrange(p1, p2, p3, nrow = 2, ncol = 2)
```


## Significant gene lists
For each of the comparisons, the number of significant probes were tabulated. Further, annotations were retrieved for all probes and only significant probes that mapped to Entrez gene IDs were written to file. These gene lists can be found here:       

1. [MEF vs iNoc](./results/MEFvsiNoc_siggenes.txt) 
2. [MEF vs DRG](./results/MEFvsDRG_siggenes.txt)  
3. [DRG vs iNoc](./results/iNocvsDRG_siggenes.txt)  

```{r annotate}
rids <- rownames(exprs(data.rma))

# Get annotations into matrix
en <- unlist(mget(rids, mogene10sttranscriptclusterENTREZID, ifnotfound=NA))
gn <- unlist(mget(rids, mogene10sttranscriptclusterGENENAME, ifnotfound=NA))
sm <- unlist(mget(rids, mogene10sttranscriptclusterSYMBOL, ifnotfound=NA))
ann <- cbind(id=rids, entrez=en, genename=gn, symbol=sm)

# Merge annotation for each comparison
annot1 <- merge(ann, MEFvsDRG, by='row.names')
annot2 <- merge(ann, MEFvsiNoc, by='row.names')
annot3 <- merge(ann, iNocvsDRG, by='row.names')
```

```{r writefile, eval=FALSE, echo=FALSE}
# Write to file
write.table(annot1[which(annot1$threshold.FDR), 3:10], file="results//MEFvsDRG_siggenes.txt", row.names=F,
            sep="\t", quote=F, append=F)
write.table(annot2[which(annot2$threshold.FDR), 3:10], file="results//MEFvsiNoc_siggenes.txt", row.names=F,
            sep="\t", quote=F, append=F)
write.table(annot3[which(annot3$threshold.FDR), 3:10], file="results//iNocvsDRG_siggenes.txt", row.names=F,
            sep="\t", quote=F, append=F)
```

### Table of significant findings
The number of significant probes and corresponding number of unique genes from each comparison are listed in the table below:
```{r, results='asis', echo=FALSE}

for (i in 1:3){
  x <- get(paste("annot", i, sep=""))
  x <- x[which(x$threshold.FDR),]
  x.genes <- unique(x$symbol)
  assign(paste("annot", i, sep=""), x.genes)
}
out <- cbind(colnames(fit2$coefficients), SigGenes, c(length(annot1), length(annot2), length(annot3)))
colnames(out) <- c("", "Significant probes", "Significant genes")
kable(out, format='markdown')
```

### Overlapping significant genes
The number of overlapping genes between the three different comparisons:
```{r image5 , echo=FALSE, fig.align='center'}
img5 <- readPNG("./results/overlappingGenes.png")
 grid.raster(img5) 
```


